### PR TITLE
Add university controller and tests

### LINKED ISSUE #15: Add University Controller and Tests
# Acceptance Criteria:

- [ ] There is <tt>UniversityController.java</tt> file in the `src/main/java/edu/ucsb/cs156/spring/backend/controllers` directory.
- [ ] <tt>UniversityController.java</tt> implements a class that implements an API endpoint that takes in the necessary
      parameters, calls the <tt>UniversityQueryService</tt>, and returns the results.
- [ ] The file  <tt>UniversityController.java</tt> file make appropriate use of the
      annotations `@Tag`, `@Operation` and `@Parameter` to document the API.
- [ ] The controller endpoint shows up on the Swagger UI page.
- [ ] There is a <tt>UniversityControllerTests.java</tt> file in the `src/test/java/edu/ucsb/cs156/spring/backend/controllers` directory.
- [ ] The tests in <tt>UniversityControllerTests</tt> provide 100% coverage of <tt>UniversityController</tt> for both jacoco and pitest.

# Details

See the assignment description at https://ucsb-cs156.github.io/s24/lab/team01 for details.

### PR DESCRIPTION
**Pull request by rainakakani21**
In this PR, we add an endpoint `/api/university/get` that can be used to get the list of universities that match a given name. 

dev deployment: http://team01-rainakakani21-dev.dokku-08.cs.ucsb.edu/swagger-ui/index.html

closes #15

### CODE REVIEW

File: src/test/java/edu/ucsb/cs156/spring/backenddemo/services/UniversityQueryServiceTests.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    @Autowired
+    private MockRestServiceServer mockRestServiceServer;
+
+    @Autowired
+    private UniversityQueryService universityQueryService;
+
+    @Test 
+    public void test_getJSON() {
+
+        String name = "Harvard";
```
Comment (Jumshim) [inline src/test/java/edu/ucsb/cs156/spring/backenddemo/services/UniversityQueryServiceTests.java:29]: Why not UCSB?

Jk, LGTM!

**Review by Jumshim — APPROVED** [review]

**Review by vaaniarora — APPROVED** [review]
nice nice lgtm!

**Review by niyati-m — APPROVED** [review]
LGTM!

**Review by emilythai — APPROVED** [review]
LGTM!

**Review by charlene-patenaude — APPROVED** [review]
lgtm :)

**Pull request merged**